src/index.js

```diff
@@ -1,38 +1,142 @@
 import './style.css';
 
-const tasks = [
-  {
-    description: 'Take out the trash',
-    completed: false,
-    index: 1,
-  },
-  {
-    description: 'Wash the dishes',
-    completed: true,
-    index: 3,
-  },
-  {
-    description: 'Buy groceries',
-    completed: false,
-    index: 2,
-  },
-];
-
-const taskList = () => {
-  let content = '';
-  tasks
-    .sort((a, b) => a.index - b.index)
-    .forEach((task) => {
-      content += `
-            <li class="task-list" id="toDo"><input type="checkbox" id="${task.description}"  value="task">${task.description}</li>
-        `;
-    });
-  return content;
+const updateLocalStorage = (toDoListArray) => {
+  localStorage.setItem('toDoList', JSON.stringify(toDoListArray));
 };
 
-const rendering = () => document.getElementById('to-dos');
+const getLocalStorage = () => {
+  const localStorageData = JSON.parse(localStorage.getItem('toDoList'));
+  let toDoListArray;
+  if (localStorageData) {
+    toDoListArray = localStorageData;
+  } else {
+    toDoListArray = [];
+  }
+  return toDoListArray;
+};
+
+const toDoTasks = getLocalStorage();
+const toDoList = document.querySelector('.toDoContainer');
+
+const addTask = (toDoListArray, task) => {
+  toDoListArray.push({ task, completed: false, id: toDoListArray.length + 1 });
+};
+
+const renderToDoList = (toDoListArray) => {
+  toDoList.innerHTML = '';
+
+  toDoListArray = getLocalStorage();
+
+  toDoListArray.forEach((toDo) => {
+    const toDoItem = document.createElement('li');
+    toDoItem.classList.add('toDoContainer-li');
+
+    const toDoCheckbox = document.createElement('input');
+    toDoCheckbox.classList.add('toDoContainer-li-checkbox');
+    toDoCheckbox.type = 'checkbox';
+    toDoCheckbox.checked = toDo.completed;
+    toDoItem.appendChild(toDoCheckbox);
+
+    const toDoText = document.createElement('input');
+    toDoText.classList.add('toDoContainer-li-text');
+    toDoText.value = toDo.task;
+    // toDoText.disabled = true;
+    toDoItem.appendChild(toDoText);
+
+    if (toDo.completed) {
+      toDoText.classList.add('completed');
+    }
+
+    const crossIcon = document.createElement('span');
+    crossIcon.classList.add('trash-can');
+    crossIcon.innerHTML = '🗑️';
+    toDoItem.appendChild(crossIcon);
+
+    toDoList.appendChild(toDoItem);
+  });
+};
+
+const editTask = (e, toDoListArray) => {
+  const clickedTask = e.target.closest('.toDoContainer-li-text');
+  clickedTask.disabled = false;
+  clickedTask.focus();
+  const taskText = clickedTask.value;
+  clickedTask.addEventListener('keypress', () => {
+    if (clickedTask.value !== '') {
+      const taskIndex = toDoListArray.findIndex(
+        (task) => task.task === taskText,
+      );
+      toDoListArray[taskIndex].task = clickedTask.value;
+      clickedTask.disabled = true;
+      updateLocalStorage(toDoListArray);
+      renderToDoList(toDoListArray);
+    }
+  });
```

Comment:
- [ ] Good job trying to fix some of the changes required by the previous reviewer. While fixing the Editing functionality you introduced a new error for editing a task as now it only accepts one letter.
You can see below how your app behaviors:
![edit](https://user-images.githubusercontent.com/83042742/213497294-7c842dc6-75eb-4c18-ac6b-15e2bd129413.gif)